design/open/hard-delete.md

```diff
@@ -0,0 +1,202 @@
+# Hard Delete Uncommitted Data
+
+## Motivation
+
+Uncommitted data which is no longer referenced (due to branch deletion, reset branch etc.) is not being deleted by lakeFS.
+This may result excessive storage usage and possible compliance issues.
+To solve this two approaches were suggested:
+1. A batch operation performed as part of an external process (GC)
+2. An online solution inside lakeFS
+
+**This document details the latter.**
```

Comment:
I might be missing some context here, but what are the reason to not consider an offline GC?

Reply:
Offline GC is also considered. We want to exhaust the options of implementing an online solution.
This document discusses only the online solutions